components/centre_power.py

```diff
@@ -30,13 +30,19 @@ def draw_base(self, canvas: Canvas):
         pass
 
     def draw_data(self, canvas: Canvas, data: Data):
-        power = data["power"]
-        rec_power = data["rec_power"]
+        power = data["power"].get()
+        rec_power = data["rec_power"].get()
 
-        power_str = f"{power:.0f}w"
-        rec_power_str = f"{rec_power:.0f}w rec"
+        if data["power"].is_valid():
+            power_str = data["power"].get_string() + "w"
+        else:
+            power_str = "--"
+        if data["rec_power"].is_valid():
+            rec_power_str = data["rec_power"].get_string() + "w rec"
+        else:
+            rec_power_str = "--"
 
-        if rec_power == 0:
+        if rec_power == 0 or rec_power is None:
```

Comment:
Sorry, should have picked this up before. Can you add `or power is None` to this? In the case that `power` is invalid but `rec_power` still has data, we'll divide `None` by a number below which python won't like.